pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -56,10 +56,18 @@ func (f *FluxAddonClient) Upgrade(ctx context.Context, managementCluster *types.
 }
 
 func FluxChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ChangeDiff {
+	if !newSpec.Cluster.IsSelfManaged() {
```

Comment:
I guess we can remove these checks now from the `Upgrade` method

Reply:
Removed